### PR TITLE
[ADDED] Allow aggregating JS metrics and service observations from multiple accounts in one config

### PR DESCRIPTION
**Pull request by piotrpio**
This PR adds the ability to have a single config (and single NATS connection) to aggregate metrics and observations from multiple accounts.

In order to achieve that, a valid set of imports/exports has to be present in configuration. This can be achieved either by stream exports from source accounts into aggregate account, or by service exports from aggregate account to source accounts. For examples, look at `test/jetstream.conf` (for advisories config) and `test/services.conf` (for service observation config)

- For `ServiceObsConfig`, new fields are added (`ExternalAccountTokenPosition` and `ExternalServiceNamePosition`) to inform surveyor where the account name and service name (optionally) will be found in subject. These have to match with the imports/exports mapping in server config

- For jetstream advisories, in order to use aggregated metrics, these subjects should be exported:
  - `$JS.EVENT.METRIC.ACC.*`, wildcard is the account name
  - `$JS.EVENT.ADVISORY.ACC.*`, wildcard is the account name

### CODE REVIEW

File: surveyor/jetstream_advisories.go [hunk 1]
```diff
@@ -420,70 +420,90 @@ func limitJSSubject(subj string) string {
 	return subj
 }
 
+const (
+	// JSAggregateMetricPrefix and JSAggregateAdvisoryPrefix are used to identify aggregate advisories
+	// When configuring exports, account name should be used as suffix to these subjects
+	JSAggregateMetricPrefix   = "$JS.EVENT.METRIC.ACC"
+	JSAggregateAdvisoryPrefix = "$JS.EVENT.ADVISORY.ACC"
```
[the remark this answers is not in the document]
Comment (piotrpio) [reply surveyor/jetstream_advisories.go:427]: Well, they would not exactly overlap, as there is the `ACC` suffix. Although if using wildcards, then then becomes a problem. So I agree, we should move the account position to be either a prefix (as you sugested) or tucked inside the advisory subject, e.g. `$JS.API.ACC.<account_name>.METRICS`. But if we do so, why do you feel we need to allow re-mapping this subject? It should be unique + it does not collide with any other `$JS.API` subject.

I'm worried about adding complexity to an already complex feature. For service observations we had to make the subject configurable, there's no way around it. But here, adding 4 more configurable fields when you would probably use the default anyway, seems excessive. Unless I'm missing a use case of course, in which case this has to be added :)

[the remark this answers is not in the document]
Comment (piotrpio) [reply surveyor/jetstream_advisories.go:427]: And you're right, this is inconsistent, I will add the ability to configure the advisories subjects

[the remark this answers is not in the document]
Comment (piotrpio) [reply surveyor/jetstream_advisories.go]: I added configuring subjects for advisories + improved validation for both advisories and observations if external config is used.

**Review by caleblloyd — COMMENTED** [review]
This looks good, my final thought is that we may want to get rid of the JSON Tags for Account Token position so that it is only exposed in library mode initially.  Then once we are happy with Library usage we can add the JSON tags back to expose to CLI mode

**Comment by piotrpio** [comment]
@caleblloyd I reverted the ability to set `external_account_config` from file.

**Review by caleblloyd — APPROVED** [review]

**Pull request merged**